_modules/vault-auth.py

```diff
@@ -44,6 +123,10 @@ def read_secret(path, key=None, url=None):
 
         vault_client = __utils__["vault_auth.get_vault_client"](url)
 
+        if not vault_client.is_authenticated():
+            log.error("Instance is not authenticated. Need to authenticate first")
+            return None
```

Comment:
This is a good place to raise an exception. The user asked to do something that cannot be done. Returning `None` hides the error in a successful exit.

Of course, then the overly broad try/except block around this whole thing _also_ needs to be reworked or removed.

Reply:
Ok, removed the broad try/except block and raised an exception